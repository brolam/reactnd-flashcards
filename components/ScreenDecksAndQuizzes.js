import React from 'react'
import { StyleSheet } from 'react-native'
import { View, FlatList, TouchableOpacity, Text } from 'react-native'
import PropTypes from 'prop-types'
import { orange, white } from '../styles/colors'
import DeckCardList from './DeckCardList'
import PanelQuizzes from './PanelQuizzes'
import { connect } from 'react-redux'
import { selectDeck, selectQuiz, setAppState, APP_STATES } from '../actions'
import { Ionicons } from '@expo/vector-icons'

const quizzesDummy = [
  { question: 'One Quetion' },
  { question: 'Two Quetion' },
  { question: 'Three Quetion' }
]


export function ScreenDecksAndQuizzes(props) {
  const { decks, selectedDeckQuizzes, dispatch } = props
  return (
    <View style={styles.container} >
      <View style={styles.containerDeckList}>
        <DeckCardList
          decks={decks}
          onSelectedOneDeck={(deck) => dispatch(selectDeck(deck.key, quizzesDummy))}
        />
        <TouchableOpacity id={'addFabButton'} style={[styles.fabButton, styles.addButton]}
          onPress={() => dispatch(setAppState(APP_STATES.NEW_DECK))}>
          <Ionicons name='md-add' size={30} color={white} />
        </TouchableOpacity>
      </View>
      <View style={styles.containerDeckQuizzes}>
        <PanelQuizzes {...props} />
      </View>
    </View>
  )
}

ScreenDecksAndQuizzes.propTypes = {
  decks: PropTypes.array.isRequired,
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1,
  },
  containerDeckList: {
    width: '40%'
  },
  containerDeckQuizzes: {
    marginRight: 20,
    marginBottom: 20,
    width: '60%'
  }, fabButton: {
    position: 'absolute',
    bottom: 20,
    right: 20,
    height: 56,
    width: 56,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 50,
  },
  addButton: {
    backgroundColor: orange,
    borderRadius: 50,
    alignItems: 'center',
    shadowColor: "#000000",
    shadowOpacity: 0.8,
    shadowRadius: 2,
    shadowOffset: {
      height: 1,
      width: 0
    },
    elevation: 4,
  },
})

function mapStateToProps(props) {
  return {
    deck: props.decks.find(deck => deck.key === props.selectedDeckKey),
    quizzes: props.selectedDeckQuizzes,
    ...props
  }
}

function mapDispatchToProps(dispatch, ownProps) {
  return {
    onStartQuiz: () => {
      dispatch(selectQuiz(0))
    },
    dispatch
  }
}

const ScreenDecksAndQuizzesConnected = connect(
  mapStateToProps,
  mapDispatchToProps
)(ScreenDecksAndQuizzes)


export default ScreenDecksAndQuizzesConnected