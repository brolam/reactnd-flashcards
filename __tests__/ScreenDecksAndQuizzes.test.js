import React from 'react'
import ScreenDecksAndQuizzes from '../components/ScreenDecksAndQuizzes'

test('renders without crashing', () => {
  const wrapper = shallow(<ScreenDecksAndQuizzes decks={deckDummies} />);
  expect(wrapper).toMatchSnapshot();
});

const deckDummies = [
  { id: 'one-item', title: 'One Deck', amountOfCards: 10 },
  { id: 'two-item', title: 'Two Deck', amountOfCards: 11 },
  { id: 'three-item',title: 'Three Deck', amountOfCards: 12 }
]