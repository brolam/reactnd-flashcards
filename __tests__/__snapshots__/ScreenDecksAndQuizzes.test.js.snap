// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "width": "40%",
      }
    }
  >
    <DeckCardList
      decks={
        Array [
          Object {
            "amountOfCards": 10,
            "key": "one-item",
            "title": "One Deck",
          },
          Object {
            "amountOfCards": 11,
            "key": "two-item",
            "title": "Two Deck",
          },
          Object {
            "amountOfCards": 12,
            "key": "three-item",
            "title": "Three Deck",
          },
        ]
      }
      onSelectedOneDeck={[Function]}
    />
    <TouchableOpacity
      activeOpacity={0.2}
      id="addFabButton"
      onPress={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "bottom": 20,
            "height": 56,
            "justifyContent": "center",
            "position": "absolute",
            "right": 20,
            "width": 56,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#f26f28",
            "borderRadius": 50,
            "elevation": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 2,
          },
        ]
      }
    >
      <Icon
        allowFontScaling={false}
        color="#fff"
        name="md-add"
        size={30}
      />
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
        "marginRight": 20,
        "width": "60%",
      }
    }
  >
    <PanelQuizzes
      decks={
        Array [
          Object {
            "amountOfCards": 10,
            "key": "one-item",
            "title": "One Deck",
          },
          Object {
            "amountOfCards": 11,
            "key": "two-item",
            "title": "Two Deck",
          },
          Object {
            "amountOfCards": 12,
            "key": "three-item",
            "title": "Three Deck",
          },
        ]
      }
      selectedDeckQuizzes={
        Array [
          Object {
            "question": "One Quetion",
          },
          Object {
            "question": "Two Quetion",
          },
          Object {
            "question": "Three Quetion",
          },
        ]
      }
    />
  </View>
</View>
`;
